### ISSUE TITLE
Selecting a Kit without a compatible cmake generator breaks all builds until restart.

### ISSUE THREAD

**Issue by phi1010** [microsoft/vscode-cmake-tools#512]
### Brief Issue Summary

The error message "[rollbar] Unhandled exception: Unhandled Promise rejection: build Error: Unable to determine CMake Generator to use {}" prevents me from building after selecting the MinGW toolchain. It also breaks Visual Studio kits in my case -- I've used the Visual Studio kits on Windows with before successfully, but after adding C:\MinGW\bin to the Path variable, the scan recognized the GCC and now refuses to build with any kit. 

### Expected:

Starting any cmake addin command (Configure, Build, Debug Target) will start cmake / the selected compiler / the target.

### Apparent Behavior:

Output tab opens in the CMake/Build view mode, and shows a single-line error message.

In the developer log, it seems as if the change kit command actually fails to switch away from GCC back to the good kits. The bottom toolbar however also doesn't update, and always shows "VisualStudio 14.0 - x86" (the last good selection) after choosing the GCC kit -- no matter what kit I choose afterwards.

Restarting VisualStudio Code fixes the issue with MSVC temporarily until I select GCC as toolchain again.

### CMake Tools Log

```
[rollbar] Unhandled exception: Unhandled Promise rejection: build Error: Unable to determine CMake Generator to use {}

```

### Developer Tools Log

```
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:1375  INFO no standard startup: not a new window
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:4103 Overwriting grammar scope name to file mapping for scope source.cs.
Old grammar file: file:///c%3A/Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/csharp/syntaxes/csharp.tmLanguage.json.
New grammar file: file:///c%3A/Users/********/.vscode/extensions/maximetinu.identical-sublime-monokai-csharp-theme-colorizer-0.10.0/syntaxes/csharp.tmLanguage.json
e.register @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:4103
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] vscode-icons is active!
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] activating extension
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] starting language server
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.096Z [debug] [rollbar] Updated Rollbar payload {"environment":"production","packageJSON":{"name":"cmake-tools","version":"1.1.1"},"client":{"code_version":"1.1.1"},"platform":"win32"}
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.286Z [debug] [kit] Reading kits file C:\Users\********\AppData\Roaming\CMakeTools\cmake-tools.json
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.287Z [debug] [kit] Not reading non-existent kits file: c:\Users\********\Documents\projects\********\.vscode\cmake-kits.json
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.496Z [info] [kit] Successfully loaded 11 kits from C:\Users\********\AppData\Roaming\CMakeTools\cmake-tools.json
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.500Z [debug] [main] Safe constructing new CMakeTools instance
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.509Z [debug] [variant] Constructing VariantManager
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.523Z [debug] [main] Constructing new CMakeTools instance
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.524Z [debug] [main] Starting CMakeTools second-phase init
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.526Z [debug] [rollbar] Checking Rollbar permissions
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:41.527Z [debug] [rollbar] Rollbar enabled?  false
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] runCommand platformio Array(1) Object
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.625Z [info] [variant] Loaded new set of variants
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.656Z [debug] [main] CMakeTools instance initialization complete.
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.657Z [debug] [main] Injecting new Kit into CMake driver
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.706Z [debug] [kit] Reading kits file C:\Users\********\AppData\Roaming\CMakeTools\cmake-tools.json
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.800Z [info] [kit] Successfully loaded 11 kits from C:\Users\********\AppData\Roaming\CMakeTools\cmake-tools.json
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.837Z [debug] [kit] Not reading non-existent kits file: c:\Users\********\Documents\projects\********\.vscode\cmake-kits.json
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.840Z [debug] [rollbar] Updated Rollbar payload {"kit":{"name":"VisualStudio.14.0 - x86","visualStudio":"VisualStudio.14.0","visualStudioArchitecture":"x86","preferredGenerator":{"name":"Visual Studio 14 2015"}}}
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.840Z [debug] [main] Injecting new Kit into CMake driver
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.851Z [debug] [extension] Configuring workspace on open  file:///c%3A/Users/********/Documents/projects/********
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.853Z [debug] [main] Run configure  
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:44.860Z [debug] [main] Saving open files before configure/build
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:45.461Z [debug] [main] Starting new CMake driver
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:45.462Z [debug] [main] Starting CMake driver
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:45.463Z [debug] [driver] CMakeDriver Kit set to VisualStudio.14.0 - x86
console.ts:136 [Extension Host] Found PIO Core 3.6.0-a.11
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:47.654Z [debug] [driver] Run _refreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:47.655Z [debug] [cms-driver] Run doRefreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:47.656Z [debug] [driver] Run _refreshExpansions cb
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:47.668Z [debug] [cms-client] Started new CMake Server instance with PID 2692
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:48.671Z [debug] [driver] Trying to detect generator supported by system
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:48.709Z [info] [cms-client] Configuring using the "Visual Studio 14 2015" CMake generator
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:48.718Z [debug] [driver] Setting new variant , Emit debug information without performing optimizations
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:48.718Z [debug] [driver] Run _refreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:48.719Z [debug] [cms-driver] Run doRefreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:48.719Z [debug] [driver] Run _refreshExpansions cb
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:48.789Z [debug] [cms-client] Started new CMake Server instance with PID 2216
console.ts:136 [Extension Host] runCommand platformio Array(7) Object
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:49.791Z [debug] [driver] Trying to detect generator supported by system
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:49.850Z [info] [cms-client] Configuring using the "Visual Studio 14 2015" CMake generator
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:50.002Z [debug] [driver] Start configure  
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:50.003Z [debug] [driver] Runnnig pre-configure checks and steps
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:50.010Z [debug] [driver] Run _refreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:50.010Z [debug] [cms-driver] Run doRefreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:50.011Z [debug] [driver] Run _refreshExpansions cb
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:49:50.086Z [info] [cmake] Selecting Windows SDK version  to target Windows 10.0.16299.
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:01.535Z [info] [cmake] The C compiler identification is MSVC 19.0.24210.0
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:05.479Z [info] [cmake] The CXX compiler identification is MSVC 19.0.24210.0
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:05.632Z [info] [cmake] Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:09.622Z [info] [cmake] Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe -- works
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:09.634Z [info] [cmake] Detecting C compiler ABI info
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:11.160Z [info] [cmake] Detecting C compiler ABI info - done
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:11.170Z [info] [cmake] Detecting C compile features
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:12.544Z [info] [cmake] Detecting C compile features - done
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:12.582Z [info] [cmake] Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:13.817Z [info] [cmake] Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe -- works
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:13.821Z [info] [cmake] Detecting CXX compiler ABI info
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:15.156Z [info] [cmake] Detecting CXX compiler ABI info - done
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:15.166Z [info] [cmake] Detecting CXX compile features
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.054Z [info] [cmake] Detecting CXX compile features - done
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.081Z [info] [cmake] Configuring done
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.175Z [info] [cmake] Generating done
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.182Z [info] [cmake] CMake Warning:
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.182Z [info] [cmake]   Manually-specified variables were not used by the project:
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.183Z [info] [cmake] 
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.183Z [info] [cmake]     CMAKE_EXPORT_COMPILE_COMMANDS
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.183Z [info] [cmake] 
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.184Z [info] [cmake] 
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.506Z [debug] [cache] Reading CMake cache file c:/users/********/documents/projects/********/build/cmakecache.txt
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:50:17.509Z [debug] [cache] Parsing CMake cache string
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.670Z [debug] [extension] [8566] cmake.configure started
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.672Z [debug] [main] Run configure  
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.673Z [debug] [main] Saving open files before configure/build
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.975Z [debug] [driver] Start configure  
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.976Z [debug] [driver] Runnnig pre-configure checks and steps
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.984Z [debug] [driver] Run _refreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.985Z [debug] [cms-driver] Run doRefreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.985Z [debug] [driver] Run _refreshExpansions cb
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:31.996Z [info] [cmake] Selecting Windows SDK version  to target Windows 10.0.16299.
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:32.040Z [info] [cmake] Configuring done
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:32.107Z [info] [cmake] Generating done
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:32.282Z [debug] [cache] Reading CMake cache file c:/users/********/documents/projects/********/build/cmakecache.txt
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:32.285Z [debug] [cache] Parsing CMake cache string
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:32.423Z [debug] [extension] [8566] cmake.configure finished (returned 0)
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:38.727Z [debug] [extension] [5224] cmake.selectKit started
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:38.728Z [debug] [extension] Start selection of kits. Found 12 kits.
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:38.729Z [debug] [extension] Opening kit selection QuickPick
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:41.104Z [debug] [extension] User selected kit  {"label":"GCC 6.3.0","description":"Using compilers: CXX = C:\\MinGW\\bin\\g++.exe, C = C:\\MinGW\\bin\\gcc.exe","kit":{"name":"GCC 6.3.0","compilers":{"CXX":"C:\\MinGW\\bin\\g++.exe","C":"C:\\MinGW\\bin\\gcc.exe"},"preferredGenerator":{"name":"Unix Makefiles","platform":"x64"}}}
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:41.105Z [debug] [rollbar] Updated Rollbar payload {"kit":{"name":"GCC 6.3.0","compilers":{"CXX":"C:\\MinGW\\bin\\g++.exe","C":"C:\\MinGW\\bin\\gcc.exe"},"preferredGenerator":{"name":"Unix Makefiles","platform":"x64"}}}
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:41.106Z [debug] [main] Injecting new Kit into CMake driver
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:41.107Z [info] [driver] Switching to kit: GCC 6.3.0
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:41.107Z [debug] [kit] Need clean: Kit changed
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:41.179Z [debug] [cms-driver] Wiping build directory
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:41.253Z [debug] [driver] CMakeDriver Kit set to GCC 6.3.0
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:41.262Z [debug] [cms-client] Started new CMake Server instance with PID 5804
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:42.275Z [debug] [driver] Trying to detect generator supported by system
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:42.339Z [error] [cms-client] None of preferred generators available on the system.
t.log @ console.ts:136
notificationsAlerts.ts:42 Unable to determine what CMake generator to use. Please install or configure a preferred generator, or update settings.json or your Kit configuration.
e.onDidNotificationChange @ notificationsAlerts.ts:42
notificationsAlerts.ts:42 Unable to determine CMake Generator to use
e.onDidNotificationChange @ notificationsAlerts.ts:42
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:42.411Z [fatal] [rollbar] Unhandled exception: Unhandled Promise rejection: selectKit Error: Unable to determine CMake Generator to use {}
t.log @ console.ts:136
console.ts:136 [Extension Host] Error: Unable to determine CMake Generator to use
	at onHello (C:\Users\********\.vscode\extensions\vector-of-bool.cmake-tools-1.1.1\out\src\cms-client.js:259:39)
	at <anonymous>
	at process._tickCallback (internal/process/next_tick.js:109:7)
t.log @ console.ts:136
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.630Z [debug] [extension] [6443] cmake.configure started
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.630Z [debug] [main] Run configure  
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.631Z [debug] [main] Saving open files before configure/build
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.873Z [debug] [driver] Start configure  
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.873Z [debug] [driver] Runnnig pre-configure checks and steps
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.874Z [debug] [driver] Using compilers in GCC 6.3.0 for configure
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.877Z [debug] [driver] Run _refreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.877Z [debug] [cms-driver] Run doRefreshExpansions
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.878Z [debug] [driver] Run _refreshExpansions cb
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:52:46.880Z [fatal] [rollbar] Unhandled exception: Unhandled Promise rejection: configure Error: Unable to determine CMake Generator to use {}
t.log @ console.ts:136
console.ts:136 [Extension Host] Error: Unable to determine CMake Generator to use
	at onHello (C:\Users\********\.vscode\extensions\vector-of-bool.cmake-tools-1.1.1\out\src\cms-client.js:259:39)
	at <anonymous>
	at process._tickCallback (internal/process/next_tick.js:109:7)
t.log @ console.ts:136
notificationsAlerts.ts:40 Error: Unable to determine CMake Generator to use
	at onHello (C:\Users\********\.vscode\extensions\vector-of-bool.cmake-tools-1.1.1\out\src\cms-client.js:259:39)
	at <anonymous>
	at process._tickCallback (internal/process/next_tick.js:109:7)
e.onDidNotificationChange @ notificationsAlerts.ts:40
console.ts:136 [Extension Host] No editors associated with document: c:\Users\********\Documents\projects\********\.vscode\settings.json
t.log @ console.ts:136
console.ts:136 [Extension Host] TextEditor is closed/disposed
t.log @ console.ts:136
console.ts:136 [Extension Host] TextEditor is closed/disposed
t.log @ console.ts:136
console.ts:136 [Extension Host] TextEditor is closed/disposed
t.log @ console.ts:136
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.010Z [debug] [extension] [2069] cmake.build started
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.013Z [debug] [main] Run build 
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.100Z [debug] [main] Saving open files before configure/build
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.103Z [debug] [main] Run configure  
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.103Z [debug] [main] Saving open files before configure/build
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.358Z [debug] [driver] Start configure  
console.ts:136 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.358Z [debug] [driver] Runnnig pre-configure checks and steps
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.359Z [debug] [driver] Using compilers in GCC 6.3.0 for configure
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.362Z [debug] [driver] Run _refreshExpansions
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.363Z [debug] [cms-driver] Run doRefreshExpansions
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.363Z [debug] [driver] Run _refreshExpansions cb
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T16:55:04.365Z [fatal] [rollbar] Unhandled exception: Unhandled Promise rejection: build Error: Unable to determine CMake Generator to use {}
t.log @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] Error: Unable to determine CMake Generator to use
	at onHello (C:\Users\********\.vscode\extensions\vector-of-bool.cmake-tools-1.1.1\out\src\cms-client.js:259:39)
	at <anonymous>
	at process._tickCallback (internal/process/next_tick.js:109:7)
t.log @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270

[...]

/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:08.451Z [debug] [extension] [6126] cmake.selectKit started
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:08.452Z [debug] [extension] Start selection of kits. Found 12 kits.
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:08.452Z [debug] [extension] Opening kit selection QuickPick
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:11.767Z [debug] [extension] User selected kit  {"label":"VisualStudio.14.0 - amd64","description":"Using compilers for VisualStudio.14.0 (amd64 architecture)","kit":{"name":"VisualStudio.14.0 - amd64","visualStudio":"VisualStudio.14.0","visualStudioArchitecture":"amd64","preferredGenerator":{"name":"Visual Studio 14 2015","platform":"x64"}}}
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:11.768Z [debug] [rollbar] Updated Rollbar payload {"kit":{"name":"VisualStudio.14.0 - amd64","visualStudio":"VisualStudio.14.0","visualStudioArchitecture":"amd64","preferredGenerator":{"name":"Visual Studio 14 2015","platform":"x64"}}}
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:11.769Z [debug] [main] Injecting new Kit into CMake driver
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:11.770Z [info] [driver] Switching to kit: VisualStudio.14.0 - amd64
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:11.772Z [debug] [kit] Need clean: Kit changed
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:11.773Z [fatal] [rollbar] Unhandled exception: Unhandled Promise rejection: selectKit Error: Unable to determine CMake Generator to use {}
t.log @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] Error: Unable to determine CMake Generator to use
	at onHello (C:\Users\********\.vscode\extensions\vector-of-bool.cmake-tools-1.1.1\out\src\cms-client.js:259:39)
	at <anonymous>
	at process._tickCallback (internal/process/next_tick.js:109:7)
t.log @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:2364 Unable to determine CMake Generator to use
e.onDidNotificationChange @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:2364
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.640Z [debug] [extension] [2575] cmake.build started
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.641Z [debug] [main] Run build 
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.731Z [debug] [main] Saving open files before configure/build
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.732Z [debug] [main] Run configure  
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.733Z [debug] [main] Saving open files before configure/build
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.931Z [debug] [driver] Start configure  
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.931Z [debug] [driver] Runnnig pre-configure checks and steps
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.932Z [debug] [driver] Using compilers in GCC 6.3.0 for configure
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.934Z [debug] [driver] Run _refreshExpansions
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.935Z [debug] [cms-driver] Run doRefreshExpansions
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.935Z [debug] [driver] Run _refreshExpansions cb
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] [CMakeTools] 2018-08-06T17:06:13.937Z [fatal] [rollbar] Unhandled exception: Unhandled Promise rejection: build Error: Unable to determine CMake Generator to use {}
t.log @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] Error: Unable to determine CMake Generator to use
	at onHello (C:\Users\********\.vscode\extensions\vector-of-bool.cmake-tools-1.1.1\out\src\cms-client.js:259:39)
	at <anonymous>
	at process._tickCallback (internal/process/next_tick.js:109:7)
t.log @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270
/C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270 [Extension Host] No editors associated with document: c:\Users\********\AppData\Roaming\Code\User\settings.json
t.log @ /C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:270

[...]
```

### Platform and Versions

- **Operating System**: Windows 10
- **CMake Version**: 3.12.0
- **VSCode Version**: 1.25.1
- **CMake Tools Extension Version**: 1.1.1
- **Compiler/Toolchain**: VS14.0 MSVC 19.0.24210.0 and MinGW.org GCC-6.3.0-1

### Other

Setting cmake.generator to "Unix Makefiles" only helps before this error state is reached; afterwards, it's completely ignored.

**Comment by vector-of-bool** [comment]
I'll take a look. Stray exceptions breaking things...

**Comment by vector-of-bool** [comment]
Sorry, but this one will have to be pushed back. 1.1.2 is long overdue and I've already closed all the ready-to-fix issues. Will push in next release.

**Comment by bobbrow** [comment]
This issue has been addressed in CMake Tools 1.2.0 which is available in the VS Code Marketplace or on the [Releases Page](https://github.com/microsoft/vscode-cmake-tools/releases/tag/v1.2.0)

Because the publisher id has changed, VS Code does not automatically upgrade you yet. The Marketplace team will enable this in the next few days.  If you want to install 1.2.0 now, please make sure you uninstall any earlier versions of CMake Tools manually to avoid any conflicts.
